servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SwitchMapSignal.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2023 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+/**
+ * A signal containing the data from a series of {@link Publisher}s switched in a serial fashion.
+ * @param <T> Type of the data.
+ * @see Publisher#switchMap(Function)
+ */
+public interface SwitchMapSignal<T> {
```

Comment:
this isn't strictly required for the operator, but some control flows require knowing if this is the first emission from a new publisher (e.g. may update state that is preserved across retry, ..)